### ISSUE TITLE
Allow actor system names with upper-case chars in Lease

### ISSUE THREAD

**Issue by DennisVDB** [akka/akka-management#839]
We have an actor system with upper-case characters that stops us from using the K8s lease. Akka cluster creates a lease with the name `TEST-akka-sbr`. We wouldn't like change the actor name, which means a downtime to redeploy the cluster from scratch.

The fix made in https://github.com/akka/akka-management/pull/812 is one step towards it working but it will then fail during the creation with: 
```
akka.coordination.lease.LeaseException: PUT for lease TEST-akka-sbr returned unexpected status code 400 Bad Request. Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the name of the object (TEST-akka-sbr) does not match the name on the URL (test-akka-sbr)","reason":"BadRequest","code":400}
```

It can be fixed by lower casing the `name` coming from the `LeaseSettings`. Basically making it case-insensitive. Does this sound like a good idea?

**Comment by DennisVDB** [comment]
Fixed by #853

**Comment by ignasi35** [comment]
@DennisVDB since Akka 2.6.13 the lease name is [configurable](https://github.com/akka/akka/pull/30049)
